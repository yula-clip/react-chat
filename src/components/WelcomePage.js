import React from "react";
import PropTypes from "prop-types";
import { Redirect } from "react-router-dom";
import { withStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Paper from "@material-ui/core/Paper";
import Grid from "@material-ui/core/Grid";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";
import LoginForm from "./LoginForm";
import SignupForm from "./SignUpForm";
import ErrorMessage from "./ErrorMessage";

const styles = theme => ({
  paper: {
    marginTop: 64 + theme.spacing.unit * 3,
    width: 500
  },
  tabContent: {
    padding: theme.spacing.unit * 3
  }
});

class WelcomePage extends React.Component {
  static propTypes = {
    classes: PropTypes.objectOf(PropTypes.string).isRequired,
    signup: PropTypes.func.isRequired,
    login: PropTypes.func.isRequired,
    recieveAuth: PropTypes.func.isRequired,
    isAuthenticated: PropTypes.bool.isRequired,
    error: PropTypes.instanceOf(Error)
  };

  static defaultProps = {
    error: null
  };

  state = {
    activeTab: 0
  };

  componentDidMount() {
    this.props.recieveAuth();
  }

  handleTabChage = (event, value) => {
    this.setState({ activeTab: value });
  };

  render() {
    const { classes, signup, login, isAuthenticated, error } = this.props;
    const { activeTab } = this.state;

    if (isAuthenticated) {
      return <Redirect to="/chat" />;
    }
    return (
      <React.Fragment>
        <AppBar>
          <Toolbar>
            <Typography variant="h6" color="inherit" style={{ flex: 1 }}>
              DogeCodes React Chat
            </Typography>
          </Toolbar>
        </AppBar>
        <Grid container justify="center">
          <Grid item>
            <Paper className={classes.paper}>
              <AppBar position="static" color="default">
                <Tabs
                  value={activeTab}
                  onChange={this.handleTabChage}
                  variant="fullWidth"
                >
                  <Tab label="Login" />
                  <Tab label="Sign Up" />
                </Tabs>
              </AppBar>
              <div className={classes.tabContent}>
                {activeTab === 0 && <LoginForm onSubmit={login} />}
                {activeTab === 1 && <SignupForm onSubmit={signup} />}
              </div>
            </Paper>
          </Grid>
        </Grid>
        <ErrorMessage error={error} />
      </React.Fragment>
    );
  }
}

export default withStyles(styles)(WelcomePage);
