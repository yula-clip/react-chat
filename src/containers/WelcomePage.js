import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import WelcomePage from '../components/WelcomePage';
import { signup, login } from '../actions'
const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  signup,
  login,
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WelcomePage);
