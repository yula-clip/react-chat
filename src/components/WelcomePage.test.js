/* eslint-env jest */
import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter, Route } from "react-router-dom";
import renderer from "react-test-renderer";
import WelcomePage from "./WelcomePage";

jest.mock("./LoginForm", () => () => "LoginForm");
jest.mock("./SignupForm", () => () => "SignupForm");
jest.mock("./ErrorMessage", () => () => "./ErrorMessage");

const mockProps = {
  signup: jest.fn(),
  login: jest.fn(),
  recieveAuth: jest.fn(),
  isAuthenticated: false,
  error: null
};

describe("<WelcomePage />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter initialEntries={["/"]}>
        <Route
          path="/(welcome)?"
          render={props => <WelcomePage {...mockProps} {...props} />}
        />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={["/"]}>
          <Route
            path="/(welcome)?"
            render={props => <WelcomePage {...mockProps} {...props} />}
          />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
