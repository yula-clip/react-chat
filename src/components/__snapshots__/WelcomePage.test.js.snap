// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomePage /> renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root-220 MuiPaper-elevation4-226 MuiAppBar-root-211 MuiAppBar-positionFixed-212 MuiAppBar-colorPrimary-218 mui-fixed"
  >
    <div
      className="MuiToolbar-root-247 MuiToolbar-regular-249 MuiToolbar-gutters-248"
    >
      <h6
        className="MuiTypography-root-251 MuiTypography-h6-268 MuiTypography-colorInherit-280"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        DogeCodes React Chat
      </h6>
    </div>
  </header>,
  <div
    className="MuiGrid-container-286 MuiGrid-justify-xs-center-303"
  >
    <div
      className="MuiGrid-item-287"
    >
      <div
        className="MuiPaper-root-220 MuiPaper-elevation2-224 MuiPaper-rounded-221 WelcomePage-paper-209"
      >
        <header
          className="MuiPaper-root-220 MuiPaper-elevation4-226 MuiAppBar-root-211 MuiAppBar-positionStatic-215 MuiAppBar-colorDefault-217"
        >
          <div
            className="MuiTabs-root-383"
          >
            <div
              className="MuiTabs-flexContainer-384"
            >
              <div
                className="MuiTabs-scroller-386 MuiTabs-fixed-387"
                onScroll={[Function]}
                role="tablist"
                style={
                  Object {
                    "marginBottom": 0,
                  }
                }
              >
                <div
                  className="MuiTabs-flexContainer-384"
                >
                  <button
                    aria-selected={true}
                    className="MuiButtonBase-root-404 MuiTab-root-392 MuiTab-textColorInherit-394 MuiTab-selected-397 MuiTab-fullWidth-399"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="tab"
                    tabIndex="0"
                    type="button"
                  >
                    <span
                      className="MuiTab-wrapper-400"
                    >
                      <span
                        className="MuiTab-labelContainer-401"
                      >
                        <span
                          className="MuiTab-label-402"
                        >
                          Login
                        </span>
                      </span>
                    </span>
                    <span
                      className="MuiTouchRipple-root-410"
                    />
                  </button>
                  <button
                    aria-selected={false}
                    className="MuiButtonBase-root-404 MuiTab-root-392 MuiTab-textColorInherit-394 MuiTab-fullWidth-399"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="tab"
                    tabIndex="0"
                    type="button"
                  >
                    <span
                      className="MuiTab-wrapper-400"
                    >
                      <span
                        className="MuiTab-labelContainer-401"
                      >
                        <span
                          className="MuiTab-label-402"
                        >
                          Sign Up
                        </span>
                      </span>
                    </span>
                    <span
                      className="MuiTouchRipple-root-410"
                    />
                  </button>
                </div>
                <span
                  className="MuiPrivateTabIndicator-root-407 MuiPrivateTabIndicator-colorSecondary-409 MuiTabs-indicator-391"
                  style={
                    Object {
                      "left": 0,
                      "width": 0,
                    }
                  }
                />
              </div>
            </div>
          </div>
        </header>
        <div
          className="WelcomePage-tabContent-210"
        >
          LoginForm
        </div>
      </div>
    </div>
  </div>,
  "./ErrorMessage",
]
`;
